pypeit/specobjs.py

```diff
@@ -766,6 +766,9 @@ def write_to_fits(self, subheader, outfile, overwrite=True, update_det=None,
                         header[key.upper()] = line
             else:
                 header[key.upper()] = subheader[key]
+                # Also store the datetime in ISOT format
+                if key.upper() == 'MJD':
+                    header['DATETIME'] = (Time(subheader[key][0], format='mjd').isot, "Date and time of the observation in ISOT format")
```

Comment:
I don't know if you noticed, but the CI tests are failing because of this. The error is:
```
                # Also store the datetime in ISOT format
                if key.upper() == 'MJD':
>                   header['DATETIME'] = (Time(subheader[key][0], format='mjd').isot, "Date and time of the observation in ISOT format")
E                   TypeError: 'float' object is not subscriptable
```

Reply:
Thanks @debora-pe! This line of code is fine for the reduction but not the flux calibration step. So, I've implemented a fix, and this should work again now. It would probably be a good idea to run the dev-suite just to be sure, and I've just started this now.

Reply:
Thanks @debora-pe! This line of code is fine for the reduction but not the flux calibration step. So, I've implemented a fix, and this should work again now. It would probably be a good idea to run the dev-suite just to be sure, and I've just started this now.